etcdserver/server.go

```diff
@@ -1002,6 +1002,26 @@ func (s *EtcdServer) LeaderStats() []byte {
 
 func (s *EtcdServer) StoreStats() []byte { return s.store.JsonStats() }
 
+func (s *EtcdServer) checkMembershipOperationPermission(ctx context.Context) error {
```

Comment:
Ahh, I misunderstood your intention. You wanted to add the nil check mainly for test code, but I thought that you were talking about the idea of detecting auth is enabled or not with `EtcdServer.authStore` is nil or not instead of calling `EtcdServer.isAuthEnabled()` (yes, if this understanding is correct, two ways of detection coexist...). Of course I agree with your idea of reducing the dummy code and add nil check to test code. Sorry for that.